client/state/purchases/reducer.js

```diff
@@ -109,65 +110,60 @@ function updatePurchases( existingPurchases, action ) {
 	return purchases;
 }
 
-export const items = ( state = initialState, action ) => {
-	const { type } = action;
-
-	switch ( type ) {
-		case PURCHASES_REMOVE:
-			return { ...state,
-				data: [],
-				hasLoadedSitePurchasesFromServer: false,
-				hasLoadedUserPurchasesFromServer: false
-			};
-		case PURCHASE_REMOVE:
-			return { ...state,
-				data: [],
-				isFetchingSitePurchases: false,
-				isFetchingUserPurchases: false
-			};
-		case PURCHASES_SITE_FETCH:
-			return { ...state, isFetchingSitePurchases: true };
-		case PURCHASES_USER_FETCH:
-			return { ...state, isFetchingUserPurchases: true };
-
-		case PURCHASE_REMOVE_COMPLETED:
-			return { ...state,
-				data: updatePurchases( state.data, action ),
-				error: null,
-				isFetchingSitePurchases: false,
-				isFetchingUserPurchases: false,
-				hasLoadedSitePurchasesFromServer: true,
-				hasLoadedUserPurchasesFromServer: true
-			};
-		case PURCHASES_SITE_FETCH_COMPLETED:
-			return { ...state,
-				data: updatePurchases( state.data, action ),
-				error: null,
-				isFetchingSitePurchases: false,
-				hasLoadedSitePurchasesFromServer: true
-			};
-		case PURCHASES_USER_FETCH_COMPLETED:
-			return { ...state,
-				data: updatePurchases( state.data, action ),
-				error: null,
-				isFetchingUserPurchases: false,
-				hasLoadedUserPurchasesFromServer: true
-			};
-
-		case PURCHASE_REMOVE_FAILED:
-		case PURCHASES_SITE_FETCH_FAILED:
-		case PURCHASES_USER_FETCH_FAILED:
-			return { ...state, error: action.error };
-
-		case PRIVACY_PROTECTION_CANCEL_COMPLETED:
-			return updatePurchaseById( state, action.purchase.id, action.purchase );
-
-		case PRIVACY_PROTECTION_CANCEL_FAILED:
-			return updatePurchaseById( state, action.purchaseId, {
-				error: action.error
-			} );
-
-		default:
-			return state;
-	}
-};
+const assignError = ( state, action ) => ( { ...state, error: action.error } );
+
+export const items = createReducer( initialState, {
+	[ PURCHASES_REMOVE ]: ( state ) => ( {
+		...state,
+		data: [],
+		hasLoadedSitePurchasesFromServer: false,
+		hasLoadedUserPurchasesFromServer: false
+	} ),
+
+	[ PURCHASE_REMOVE ]: ( state ) => ( {
+		...state,
```

Comment:
i guess it's because we shouldn't reset the list of purchases when we fire `PURCHASE_REMOVE`. We should keep the list of previous purchases until this reducer receives `PURCHASE_REMOVE_COMPLETED`.